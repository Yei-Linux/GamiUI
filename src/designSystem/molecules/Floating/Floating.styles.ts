import styled from "@emotion/styled";
import {
  floatingBottom,
  floatingLeft,
  floatingRight,
  floatingTop,
} from "../../../styles/utilities/floating";
import { twinStyles } from "../../../styles/utilities/twinStyles";

export const FloatingWrapper = styled.div<{
  direction: "left" | "top" | "right" | "bottom";
  position: string;
  border?: string;
  shadow?: string;
  width?: string;
  heigth?: string;
}>`
  width: fit-content;
  padding: 1rem;

  position: absolute;
  transition: all 1.5s;

  ${(props: any) => twinStyles(props)};

  ${(props) => props.direction == "right" && floatingRight(props)}
  ${(props) => props.direction == "left" && floatingLeft(props)}
  ${(props) => props.direction == "top" && floatingTop(props)}
  ${(props) => props.direction == "bottom" && floatingBottom(props)}
`;

export const FloatingHeader = styled.div`
  display: flex;
`;
