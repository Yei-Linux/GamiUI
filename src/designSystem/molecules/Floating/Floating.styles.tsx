import { css } from '@emotion/react'
import styled from '@emotion/styled'
import { BorderType, HeightType, ShadowType, WidthType } from 'core/domain/types'
import { zIndex } from 'styles/tokens'
import { mediaQuery } from 'styles/utilities/breakpoints'
import { setGenericPropStyles } from 'styles/utilities/genericPropStyles'

export const floatingRight = () => css`
  margin: auto;
  right: 10px;

  ${mediaQuery('md')} {
    top: 40%;
  }
`

export const floatingLeft = () => css`
  margin: auto;
  left: 10px;

  ${mediaQuery('md')} {
    top: 40%;
  }
`

export const floatingTop = () => css`
  margin: auto;
  top: 10px;

  ${mediaQuery('md')} {
    left: 40%;
  }
`

export const floatingBottom = () => css`
  margin: auto;
  bottom: 10px;

  ${mediaQuery('md')} {
    left: 40%;
  }
`

export const Floating = styled.div<{
  $direction: 'left' | 'top' | 'right' | 'bottom'
  $border?: BorderType
  $shadow?: ShadowType
  $width?: WidthType
  $heigth?: HeightType
}>`
  z-index: ${zIndex[5]};
  background-color: white;
  width: fit-content;
  overflow: hidden;
  padding: 1rem;

  position: fixed;

  ${({ $border, $shadow, $width, $heigth }) =>
    setGenericPropStyles({
      border: $border || 'sm',
      shadow: $shadow || 'sm',
      width: $width || 'auto',
      heigth: $heigth || 'auto',
    })};

  ${({ $direction }) => $direction == 'right' && floatingRight()}
  ${({ $direction }) => $direction == 'left' && floatingLeft()}
  ${({ $direction }) => $direction == 'top' && floatingTop()}
  ${({ $direction }) => $direction == 'bottom' && floatingBottom()}
`

export const FloatingHeader = styled.div`
  display: flex;
`
