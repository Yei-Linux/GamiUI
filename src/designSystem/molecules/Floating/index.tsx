export { default } from "./Floating";
export { options } from "./constants";
