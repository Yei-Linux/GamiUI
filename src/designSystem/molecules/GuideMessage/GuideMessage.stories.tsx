import { Meta } from "@storybook/react";
import { getTemplate, stylesControl } from "../../../core/helpers/storybook";
import { options as optionsGuide } from "./constants";

import GuideMessage from ".";

const Template = getTemplate(GuideMessage);

export default {
  title: "Molecules/GuideMessage",
  component: GuideMessage,
  args: { ...GuideMessage.defaultProps },
  argTypes: {
    message: { control: { type: "text" } },
    guideType: { control: { type: "select", options: optionsGuide.type } },
    ...stylesControl,
  },
} as Meta;

export const Left = Template.bind({});
Left.args = {
  guideType: "run",
  message: "Left Direction",
};

export const Right = Template.bind({});
Right.args = {
  guideType: "skater",
  direction: "right",
  message: "Right Direction",
};
