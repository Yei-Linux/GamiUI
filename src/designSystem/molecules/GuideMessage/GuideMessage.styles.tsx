import styled from "@emotion/styled";
import tw from "twin.macro";
import { twinStyles } from "../../../styles/utilities/twinStyles";

export const GuideMessageWrapper = styled.div<{
  border?: string;
  shadow?: string;
  width?: string;
  heigth?: string;
  direction: "left" | "right";
}>`
  ${tw`flex justify-center items-center`}

  ${(props: any) => twinStyles(props)};
  flex-direction: ${(props) =>
    props.direction == "left" ? "row" : "row-reverse"};
`;
