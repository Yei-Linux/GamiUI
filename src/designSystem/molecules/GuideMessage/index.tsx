export { default } from "./GuideMessage";
export { options } from "./constants";
