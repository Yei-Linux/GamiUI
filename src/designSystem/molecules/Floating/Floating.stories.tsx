import { Meta } from "@storybook/react";
import { stylesControl } from "../../../core/helpers/storybook";

import Floating from ".";
import React from "react";
import GuideMessage from "../GuideMessage";

export default {
  title: "Molecules/Floating",
  component: Floating,
  args: { ...Floating.defaultProps },
  argTypes: {
    ...stylesControl,
  },
} as Meta;

export const Right = (args: any) => (
  <Floating {...args}>
    <GuideMessage
      guideType="chill"
      message="Toma un descanso!"
      direction="right"
      background="#ff75a0"
    />
  </Floating>
);

Right.args = {
  direction: "right",
};

export const Left = (args: any) => (
  <Floating {...args}>
    <GuideMessage
      guideType="chill"
      message="Toma un descanso!"
      direction="right"
      background="#ff75a0"
    />
  </Floating>
);

Left.args = {
  direction: "left",
};

export const Top = (args: any) => (
  <Floating {...args}>
    <GuideMessage
      guideType="chill"
      message="Toma un descanso!"
      direction="right"
      background="#ff75a0"
    />
  </Floating>
);

Top.args = {
  direction: "top",
};

export const Bottom = (args: any) => (
  <Floating {...args}>
    <GuideMessage
      guideType="chill"
      message="Toma un descanso!"
      direction="right"
      background="#ff75a0"
    />
  </Floating>
);

Bottom.args = {
  direction: "bottom",
};
