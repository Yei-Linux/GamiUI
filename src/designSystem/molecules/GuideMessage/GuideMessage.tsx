import React from "react";
import { IGeneralProps } from "../../../core/domain/interfaces/IGeneralProps";
import { GuidesNames } from "../../../core/domain/types";
import AnimatorGuide from "../../atoms/AnimatorGuide/AnimatorGuide";
import Message from "../../atoms/Message/Message";
import { GuideMessageWrapper } from "./GuideMessage.styles";

export interface GuideMessageProps extends IGeneralProps {
  /**
   * Guide Type
   */
  guideType: GuidesNames;
  /**
   * Message Text
   */
  message: string;
  /**
   * Message Direction
   */
  direction: "left" | "right";
  /**
   * Background
   */
  background: string;
}

const GuideMessage = ({
  guideType,
  message,
  direction,
  background,
  ...args
}: GuideMessageProps) => (
  <GuideMessageWrapper direction={direction} {...args}>
    <AnimatorGuide type={guideType} />
    <Message
      style={{ marginTop: "15px" }}
      text={message}
      direction={direction}
      background={background}
    />
  </GuideMessageWrapper>
);

GuideMessage.defaultProps = {
  direction: "left",
  guideType: "singer",
  background: "#7868e6",
};

export default GuideMessage;
