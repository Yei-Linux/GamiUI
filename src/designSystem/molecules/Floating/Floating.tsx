import React, { useRef, useState } from "react";
import { IGeneralProps } from "../../../core/domain/interfaces/IGeneralProps";
import Icon from "../../atoms/Icon";
import { FloatingHeader, FloatingWrapper } from "./Floating.styles";

export interface FloatingProps extends IGeneralProps {
  children: React.ReactNode;
  direction: "left" | "top" | "right" | "bottom";
}

const Floating = ({ children, direction, ...args }: FloatingProps) => {
  const refFloating: any = useRef();
  const [position, setPosition] = useState("0px");

  return (
    <FloatingWrapper
      ref={refFloating}
      position={position}
      direction={direction}
      {...args}
    >
      <FloatingHeader>
        <Icon
          fill="#7868e6"
          name="close"
          size="15px"
          onClick={() => {
            setPosition(`-${refFloating?.current.clientWidth.toString()}px`);
          }}
        />
      </FloatingHeader>

      {children}
    </FloatingWrapper>
  );
};

Floating.defaultProps = {
  direction: "right",
  shadow: "MEDIUM",
};

export default Floating;
