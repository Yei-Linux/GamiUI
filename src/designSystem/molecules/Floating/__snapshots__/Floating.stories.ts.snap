// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots Molecules/Floating Bottom 1`] = `
<div
  className="css-1vk7518-FloatingWrapper ehobp2h0"
  direction="bottom"
>
  <div
    className="css-1dhy0qo-FloatingHeader ehobp2h1"
  >
    <i
      className="css-6tw2qr-IconWrapper e1cr9qb30"
      onClick={[Function]}
    >
      <svg
        fill="#7868e6"
        height="15px"
        viewBox="0 0 329.26933 329"
        width="15px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="m194.800781 164.769531 128.210938-128.214843c8.34375-8.339844 8.34375-21.824219 0-30.164063-8.339844-8.339844-21.824219-8.339844-30.164063 0l-128.214844 128.214844-128.210937-128.214844c-8.34375-8.339844-21.824219-8.339844-30.164063 0-8.34375 8.339844-8.34375 21.824219 0 30.164063l128.210938 128.214843-128.210938 128.214844c-8.34375 8.339844-8.34375 21.824219 0 30.164063 4.15625 4.160156 9.621094 6.25 15.082032 6.25 5.460937 0 10.921875-2.089844 15.082031-6.25l128.210937-128.214844 128.214844 128.214844c4.160156 4.160156 9.621094 6.25 15.082032 6.25 5.460937 0 10.921874-2.089844 15.082031-6.25 8.34375-8.339844 8.34375-21.824219 0-30.164063zm0 0"
        />
      </svg>
    </i>
  </div>
  <div
    className="css-v95ugg-GuideMessageWrapper ewha4wa0"
    direction="right"
  >
    <div
      className="css-fhxv9x-AnimatorGuideImg e9zrumb0"
    />
    <div
      className="css-15xp5fn-MessageWrapper e6z3dsf0"
      direction="right"
      style={
        Object {
          "marginTop": "15px",
        }
      }
    >
      Toma un descanso!
    </div>
  </div>
</div>
`;

exports[`Storyshots Molecules/Floating Left 1`] = `
<div
  className="css-115g4fk-FloatingWrapper ehobp2h0"
  direction="left"
>
  <div
    className="css-1dhy0qo-FloatingHeader ehobp2h1"
  >
    <i
      className="css-6tw2qr-IconWrapper e1cr9qb30"
      onClick={[Function]}
    >
      <svg
        fill="#7868e6"
        height="15px"
        viewBox="0 0 329.26933 329"
        width="15px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="m194.800781 164.769531 128.210938-128.214843c8.34375-8.339844 8.34375-21.824219 0-30.164063-8.339844-8.339844-21.824219-8.339844-30.164063 0l-128.214844 128.214844-128.210937-128.214844c-8.34375-8.339844-21.824219-8.339844-30.164063 0-8.34375 8.339844-8.34375 21.824219 0 30.164063l128.210938 128.214843-128.210938 128.214844c-8.34375 8.339844-8.34375 21.824219 0 30.164063 4.15625 4.160156 9.621094 6.25 15.082032 6.25 5.460937 0 10.921875-2.089844 15.082031-6.25l128.210937-128.214844 128.214844 128.214844c4.160156 4.160156 9.621094 6.25 15.082032 6.25 5.460937 0 10.921874-2.089844 15.082031-6.25 8.34375-8.339844 8.34375-21.824219 0-30.164063zm0 0"
        />
      </svg>
    </i>
  </div>
  <div
    className="css-v95ugg-GuideMessageWrapper ewha4wa0"
    direction="right"
  >
    <div
      className="css-fhxv9x-AnimatorGuideImg e9zrumb0"
    />
    <div
      className="css-15xp5fn-MessageWrapper e6z3dsf0"
      direction="right"
      style={
        Object {
          "marginTop": "15px",
        }
      }
    >
      Toma un descanso!
    </div>
  </div>
</div>
`;

exports[`Storyshots Molecules/Floating Right 1`] = `
<div
  className="css-1u13wgd-FloatingWrapper ehobp2h0"
  direction="right"
>
  <div
    className="css-1dhy0qo-FloatingHeader ehobp2h1"
  >
    <i
      className="css-6tw2qr-IconWrapper e1cr9qb30"
      onClick={[Function]}
    >
      <svg
        fill="#7868e6"
        height="15px"
        viewBox="0 0 329.26933 329"
        width="15px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="m194.800781 164.769531 128.210938-128.214843c8.34375-8.339844 8.34375-21.824219 0-30.164063-8.339844-8.339844-21.824219-8.339844-30.164063 0l-128.214844 128.214844-128.210937-128.214844c-8.34375-8.339844-21.824219-8.339844-30.164063 0-8.34375 8.339844-8.34375 21.824219 0 30.164063l128.210938 128.214843-128.210938 128.214844c-8.34375 8.339844-8.34375 21.824219 0 30.164063 4.15625 4.160156 9.621094 6.25 15.082032 6.25 5.460937 0 10.921875-2.089844 15.082031-6.25l128.210937-128.214844 128.214844 128.214844c4.160156 4.160156 9.621094 6.25 15.082032 6.25 5.460937 0 10.921874-2.089844 15.082031-6.25 8.34375-8.339844 8.34375-21.824219 0-30.164063zm0 0"
        />
      </svg>
    </i>
  </div>
  <div
    className="css-v95ugg-GuideMessageWrapper ewha4wa0"
    direction="right"
  >
    <div
      className="css-fhxv9x-AnimatorGuideImg e9zrumb0"
    />
    <div
      className="css-15xp5fn-MessageWrapper e6z3dsf0"
      direction="right"
      style={
        Object {
          "marginTop": "15px",
        }
      }
    >
      Toma un descanso!
    </div>
  </div>
</div>
`;

exports[`Storyshots Molecules/Floating Top 1`] = `
<div
  className="css-16nb2re-FloatingWrapper ehobp2h0"
  direction="top"
>
  <div
    className="css-1dhy0qo-FloatingHeader ehobp2h1"
  >
    <i
      className="css-6tw2qr-IconWrapper e1cr9qb30"
      onClick={[Function]}
    >
      <svg
        fill="#7868e6"
        height="15px"
        viewBox="0 0 329.26933 329"
        width="15px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="m194.800781 164.769531 128.210938-128.214843c8.34375-8.339844 8.34375-21.824219 0-30.164063-8.339844-8.339844-21.824219-8.339844-30.164063 0l-128.214844 128.214844-128.210937-128.214844c-8.34375-8.339844-21.824219-8.339844-30.164063 0-8.34375 8.339844-8.34375 21.824219 0 30.164063l128.210938 128.214843-128.210938 128.214844c-8.34375 8.339844-8.34375 21.824219 0 30.164063 4.15625 4.160156 9.621094 6.25 15.082032 6.25 5.460937 0 10.921875-2.089844 15.082031-6.25l128.210937-128.214844 128.214844 128.214844c4.160156 4.160156 9.621094 6.25 15.082032 6.25 5.460937 0 10.921874-2.089844 15.082031-6.25 8.34375-8.339844 8.34375-21.824219 0-30.164063zm0 0"
        />
      </svg>
    </i>
  </div>
  <div
    className="css-v95ugg-GuideMessageWrapper ewha4wa0"
    direction="right"
  >
    <div
      className="css-fhxv9x-AnimatorGuideImg e9zrumb0"
    />
    <div
      className="css-15xp5fn-MessageWrapper e6z3dsf0"
      direction="right"
      style={
        Object {
          "marginTop": "15px",
        }
      }
    >
      Toma un descanso!
    </div>
  </div>
</div>
`;
